concepts/admin.py

```diff
@@ -20,30 +23,49 @@ def merge_concepts(modeladmin, request, queryset):
     :param queryset:
     :return:
     """
-    
-    opts = modeladmin.model._meta
-    app_label = opts.app_label
+    #return render_to_response('admin/Sample.html')
     RESOLVED_STATUS = Concept.RESOLVED
-    resolvedConceptCount = queryset.filter(concept_state = RESOLVED_STATUS).count()
 
-    #VGNWB-121 gets called only when there is only one resolved concept
-    if resolvedConceptCount == 1:
-        resolvedConceptsList = queryset.filter(concept_state = RESOLVED_STATUS).values()
-        unResolvedConceptsList = queryset.exclude(concept_state = RESOLVED_STATUS).values()
 
-        #As there will be only one element in the list
+    if 'post' in request.POST:
+            # process the queryset here
+
+        unResolvedConceptsList = queryset.exclude(concept_state = RESOLVED_STATUS).values()
+        resolvedConceptsList = queryset.filter(concept_state = RESOLVED_STATUS).values()
         resolvedConcept = resolvedConceptsList[0]
-        resolvedConceptName = resolvedConcept['label']
 
-        context = {
-        "resolvedConcept": resolvedConceptName,
-        "opts": opts,
-        "app_label": app_label,
-        "unResolvedConcepts": unResolvedConceptsList
-        }
-        return render_to_response('admin/merge_concepts_resolved.html', context)
+        for i in range(0,len(unResolvedConceptsList)):
+            presentConcept = unResolvedConceptsList.get(i)
+            presentConcept.merged_with = resolvedConcept
+            presentConcept.concept_state = Concept.REJECTED
+
+        return render_to_response('admin/Sample.html')
+    else:
+        opts = modeladmin.model._meta
+        app_label = opts.app_label
+
+        resolvedConceptCount = queryset.filter(concept_state = RESOLVED_STATUS).count()
```

Comment:
Why define a new variable ``resolvedConceptCount`` and a new queryset here? If you define the queryset ``resolvedConceptsList`` above the ``if`` block, you can just use  ``resolvedConceptsList.count()`` later on.